velox/type/Conversions.h

```diff
@@ -59,11 +59,60 @@ struct Converter {
   }
 };
 
-/// Presto compatible cast of strings to boolean. There is a set of strings
-/// allowed to be casted to boolean. These strings are `t, f, 1, 0, true, false`
-/// and their upper case equivalents. Casting from other strings to boolean
-/// throws.
-Expected<bool> castToBoolean(const char* data, size_t len);
+/// This function casts a string to a boolean value. There is a set of strings
+/// allowed to be casted to boolean. Casting from other strings to boolean
+/// throws. By default, Presto compatible cast is conducted. The allowed strings
+/// are `t, f, 1, 0, true, false` and their upper case equivalents.
+/// @tparam TPolicy The specified cast policy. When set as SparkCastPolicy, the
+/// allowed strings are `t, f, 1, 0, y, n, true, false, yes, no` and their upper
+/// case equivalents.
+template <typename TPolicy>
+Expected<bool> castToBoolean(const char* data, size_t len) {
+  const auto& TU = static_cast<int (*)(int)>(std::toupper);
+
+  if (len == 1) {
+    auto character = TU(data[0]);
+    if (character == 'T' || character == '1') {
+      return true;
+    }
+    if (character == 'F' || character == '0') {
+      return false;
+    }
+    if constexpr (std::is_same_v<TPolicy, SparkCastPolicy>) {
+      if (character == 'Y') {
+        return true;
+      }
+      if (character == 'N') {
+        return false;
+      }
+    }
+  }
+
+  if ((len == 4) && (TU(data[0]) == 'T') && (TU(data[1]) == 'R') &&
+      (TU(data[2]) == 'U') && (TU(data[3]) == 'E')) {
+    return true;
+  }
+
+  if ((len == 5) && (TU(data[0]) == 'F') && (TU(data[1]) == 'A') &&
+      (TU(data[2]) == 'L') && (TU(data[3]) == 'S') && (TU(data[4]) == 'E')) {
+    return false;
+  }
+
+  if constexpr (std::is_same_v<TPolicy, SparkCastPolicy>) {
+    if ((len == 3) && (TU(data[0]) == 'Y') && (TU(data[1]) == 'E') &&
+        (TU(data[2]) == 'S')) {
+      return true;
+    }
+
+    if ((len == 2) && (TU(data[0]) == 'N') && (TU(data[1]) == 'O')) {
+      return false;
+    }
+  }
+
+  const std::string errorMessage =
+      fmt::format("Cannot cast {} to BOOLEAN", StringView(data, len));
+  return folly::makeUnexpected(Status::UserError(errorMessage));
```

Comment:
UserError takes a string + args, so no need to call fmt::format

```
return folly::makeUnexpected(Status::UserError("Cannot cast {} to BOOLEAN", std::string_view(data, len));
```

Reply:
Fixed. Thanks.